File: src/main/java/ru/donz/wf/behaviors/FlowProtocol.java
package ru.donz.wf.behaviors;

import akka.actor.typed.ActorRef;
import lombok.Data;

public interface FlowProtocol {

    @Data
    class IncomingExternalMessage implements FlowProtocol{
        private final String message;
        private final ActorRef<String> replyTo;
    }

    @Data
    class C01 implements FlowProtocol{
        private final String phoneNumber;
    }
}

File: src/main/kotlin/ru/onix/wf/behaviors/FlowProtocol.kt
package ru.onix.wf.behaviors

import akka.actor.typed.ActorRef

interface FlowProtocol {
    data class IncomingExternalMessage(
        val message: String,
        val replyTo: ActorRef<String>,
    ) : FlowProtocol

    data class C01(
        val phoneNumber: String,
    ) : FlowProtocol
}